stochss/handlers/util/parameter_sweep_1d.py

```diff
@@ -143,26 +144,69 @@ def get_plotly_traces(self, keys):
         return trace_list
 
 
-    def plot(self, keys=None):
+    # def plot(self, keys=None):
+    #     '''
+    #     Plot the results based on the keys using matplotlib
+
+    #     Attributes
+    #     ----------
+    #     key : list
+    #         Identifiers for the results data
+    #     '''
+    #     if len(keys) > 2:
+    #         results = self.results[keys[0]][keys[1]][keys[2]]
+    #     else:
+    #         results = self.results[keys[0]][keys[1]]
+
+    #     matplotlib.pyplot.subplots(figsize=(8, 8))
+    #     matplotlib.pyplot.title(f"Parameter Sweep - Variable: {keys[0]}")
+    #     matplotlib.pyplot.errorbar(self.param['range'], results[:, 0], results[:, 1])
+    #     matplotlib.pyplot.xlabel(self.param['parameter'],
+    #                              fontsize=16, fontweight='bold')
+    #     matplotlib.pyplot.ylabel("Population", fontsize=16, fontweight='bold')
+
+
+    @classmethod
+    def plot(cls, results, species, param, mapper="final", reducer="avg"):
         '''
-        Plot the results based on the keys using matplotlib
+        Plot the results with error bar from time series results.
 
         Attributes
         ----------
-        key : list
-            Identifiers for the results data
+        results : list
+            List of GillesPy2 results objects.
+        species : str
+            Species of interest name.
+        param : dict
+            StochSS sweep parameter dictionary.
+        mapper : str
+            Key indicating the feature extraction function to use.
+        reducer : str
+            Key indicating the ensemble aggragation function to use.
         '''
-        if len(keys) > 2:
-            results = self.results[keys[0]][keys[1]][keys[2]]
+        func_map = {"min": numpy.min, "max": numpy.max, "avg": numpy.mean,
+                    "var": numpy.var, "final": lambda res: res[-1]}
+        map_results = [[func_map[mapper](traj[species]) for traj in result] for result in results]
+        if len(map_results[0]) > 1:
+            data = [[func_map[reducer](map_result),
+                     numpy.std(map_result)] for map_result in map_results]
+            visible = True
         else:
-            results = self.results[keys[0]][keys[1]]
+            data = [[map_result[0], 0] for map_result in map_results]
+            visible = False
+        data = numpy.array(data)
+
+        error_y = dict(type="data", array=data[:, 1], visible=visible)
+        trace_list = [plotly.graph_objs.Scatter(x=param['range'],
+                                                y=data[:, 0], error_y=error_y)]
+
+        title = f"<b>Parameter Sweep - Variable: {species}</b>"
```

Comment:
Same thing here: inline species name getting returned to the client is a XSS vulnerability